package.json

```diff
@@ -17,49 +17,83 @@
     "./package.json": "./package.json"
   },
   "files": [
-    "dist/**/*",
-    "esm/**/*",
-    "*.d.ts"
+    "dist"
   ],
   "publishConfig": {
     "access": "public",
     "main": "./dist/index.js",
+    "module": "./dist/index.mjs",
     "types": "./dist/index.d.ts",
     "exports": {
       ".": {
-        "types": "./dist/index.d.ts",
-        "import": "./esm/index.mjs",
-        "require": "./dist/index.js"
+        "import": {
+          "types": "./dist/index.d.mts",
```

Comment:
*.d.mts is generated by tsup default

https://github.com/toss/es-toolkit/pull/16#discussion_r1624466058